protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1459,88 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueType(headerType string, defaultValue string) error {
+	switch headerType {
+	case
+		"string":
+		if !(isString(defaultValue)) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q, or is not properly quoted with escaped quotations", defaultValue, headerType)
+		}
+	case
+		"number":
+		_, err := strconv.ParseFloat(defaultValue, 64)
```

Comment:
I know I suggested this, but I don't think this is actually right. For example, "NaN" and "infinity" are valid floating point numbers, but they are not JSON numbers. See https://tools.ietf.org/html/rfc4627 for the exact definition of a JSON number. We may need to be more clever here, though I expect just singling out infinity/NaN might be enough.

Reply:
From the provided link, the only exclusions appear to be infinity/NaN

Thinking of something like this:
```
func isJSONNumber(s string, t string) error {
	// Numeric values that cannot be represented as sequences of digits (such as Infinity and NaN) are not permitted.
	// See: https://tools.ietf.org/html/rfc4627#section-2.4
	switch s {
	case
		"nan",
		"inf",
		"infinity":
		return fmt.Errorf("the provided number %q is not a valid JSON number", s)
	}
	_, err := strconv.ParseFloat(s, 64)
	if err != nil {
		return fmt.Errorf("the provided default value %q does not match provider type %q", s, t)
	}
	return nil
}
```